lib/rubygems/gemcutter_utilities.rb

```diff
@@ -309,15 +310,36 @@ def pretty_host(host)
   end
 
   def get_scope_params(scope)
-    scope_params = {}
+    scope_params = { index_rubygems: true }
 
     if scope
       scope_params = { scope => true }
     else
-      say "Please select scopes you want to enable for the API key (y/n)"
-      API_SCOPES.each do |s|
-        selected = ask_yes_no(s.to_s, false)
-        scope_params[s] = true if selected
+      say "The default access scope is:"
+      scope_params.each do |k, _v|
+        say "  #{k}: y"
+      end
+      say "\n"
+      customise = ask_yes_no("Do you want to customise scopes?", false)
+      if customise
+        scope_params = {}
+        exclusive_scope = nil
+        EXCLUSIVELY_API_SCOPES.each do |s|
+          selected = ask_yes_no(s.to_s, false)
+          exclusive_scope = s if selected
+          break if selected
+        end
+        if exclusive_scope
+          say "You selected #{exclusive_scope}, which must be enabled exclusively. Other scopes will be disabled."
+          keep_selection = ask_yes_no("Is this fine? (type no for selecting other scopes)", true)
+          scope_params = { exclusive_scope => true } if keep_selection
+        end
+        if exclusive_scope.nil? || (exclusive_scope && !keep_selection)
+          API_SCOPES.each do |s|
+            selected = ask_yes_no(s.to_s, false)
+            scope_params[s] = true if selected
+          end
+        end
```

Comment:
How about this alternative implementation?

```diff
diff --git a/lib/rubygems/gemcutter_utilities.rb b/lib/rubygems/gemcutter_utilities.rb
index 9b50b55ede..1236861eac 100644
--- a/lib/rubygems/gemcutter_utilities.rb
+++ b/lib/rubygems/gemcutter_utilities.rb
@@ -322,23 +322,23 @@ def get_scope_params(scope)
       say "\n"
       customise = ask_yes_no("Do you want to customise scopes?", false)
       if customise
-        scope_params = {}
-        exclusive_scope = nil
         EXCLUSIVELY_API_SCOPES.each do |s|
           selected = ask_yes_no(s.to_s, false)
-          exclusive_scope = s if selected
-          break if selected
-        end
-        if exclusive_scope
-          say "You selected #{exclusive_scope}, which must be enabled exclusively. Other scopes will be disabled."
+          next unless selected
+
+          say "You selected #{s}, which must be enabled exclusively. Other scopes will be disabled."
           keep_selection = ask_yes_no("Is this fine? (type no for selecting other scopes)", true)
-          scope_params = { exclusive_scope => true } if keep_selection
+
+          return { s => true } if keep_selection
+
+          break
         end
-        if exclusive_scope.nil? || (exclusive_scope && !keep_selection)
-          API_SCOPES.each do |s|
-            selected = ask_yes_no(s.to_s, false)
-            scope_params[s] = true if selected
-          end
+
+        scope_params = {}
+
+        API_SCOPES.each do |s|
+          selected = ask_yes_no(s.to_s, false)
+          scope_params[s] = true if selected
         end
       end
       say "\n"
```

Seems more straightforward to me. If an exclusive scope is selected, return it if it's fine, or directly skip to non exclusive scopes if it's not.

Alternatively, we could inform of exclusivity while asking, so we don't need yet another prompt?

```
The default access scope is:
  index_rubygems: y

Do you want to customise scopes? [yN]  y
show_dashboard (answering yes disables other scopes) [yN]  N
index_rubygems [yN]  N
push_rubygem [yN]  y
...
```

Reply:
thanks!

I think the alternative makes sense as well...
I can update to that if we think its better.. let me know...

Reply:
I think it's a bit simpler and usable, but let's check with @martinemde before you implement anything.

Reply:
I like it, @deivid-rodriguez. Saying ahead of time what will happen by choosing show_dashboard will produce less mistakes. I'm making a similar change to the web UI so that show_dashboard is presented at the top with the heading "Exclusive Scopes". On the CLI the best we can do is explain it up front.

Btw, no where do we explain _why_ this is the case. What are the scopes? Do we need to add a `?/h` option that prints a brief explanation of the scope?

Reply:
this:

`show_dashboard (answering yes disables other scopes)`

or

`show_dashboard (exclusive scope, answering yes will not prompt for other scopes)`

?

Reply:
The latter seems more clear to me :+1:.

Reply:
updated 🙂

Reply:
sorry for the back and forwards, I should have waited for more discussions on the issue itself.. was just looking to contribute to something... 

If theres something similar already implemented for adding a `? /h option`, or some guidance on how to do that, I can try a PR for that as well...  thanks...

Reply:
Yeah, this is great.

And don't worry, we always go back and forth on stuff like this. I appreciate your patience! 😀